src/ethereum/utils/ensure.py

```diff
@@ -12,17 +12,11 @@
 Functions that simplify checking assertions and raising exceptions.
 """
 
-from typing import Callable
-
-
-class EnsureError(Exception):
-    """
-    Represents a failed check.
-    """
+from typing import Callable, Union
 
 
 def ensure(
-    value: bool, exception_class: Callable[[], Exception] = EnsureError
+    value: bool, exception: Union[Callable[[], BaseException], BaseException]
```

Comment:
This is to avoid `ensure(..., lambda: Foo("msg..."))`?

Reply:
This is actually wrong, it should be `Union[Type[BaseException], BaseException]`.

The point is to make the semantics of `ensure()` identical to [`raise`](https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement), which takes either a subclass or instance of `BaseException`.